README.md

```diff
@@ -1,75 +1,293 @@
+* Note the `.WithSequentialIds()` method, which will likely be useful for any Id-based properties.
+  * All it does is call `.WithOneOf(<propSelector>, Enumerable.Range(1, int.MaxValue))`.
+* If you have a database object with properties representing DB relations, where there is a FK int property AND a FK object property (and possibly also a collection property on the other end of the relationship), then you probably want to use `.WithPostBuildSetup()` to ensure that everything gets suitably wired up at the end, to account for later modifications applied to the Builder.
+* You can use the `.WithBuilt/Builder()` methods to setup complex sub-properties, for which you've already defined `Builder`s.
+  * The process for finding the "default" builder for `T`s, is to find all `public static getters` on classes tagged with `[BuilderFactory]`, and then return the one named `New`, or a random one if none are named `New`.
+  * So if you want to define multiple `Builder<T>` properties AND use "default" builders, make sure you call one `New`.
+  * Equally if you're not using "default" builders, then there's no need to include the `[BuilderFactory]` attribute.
+  * If no defined `Builder` is found, then the "default" `Builder` is just one that calls `new()` on its target.
+* `Builder` objects are immutable; each method call returns a *new* `Builder` object, leaving the original one unchanged.
+
+### Setup Methods
+
+Please examine the XML docs for full details. However, in simplified form, we have:
+
+* ##### With()
+  * Set a property to a value.
+* ##### WithOneOf()
+  * Provide multiple values, and the builder will cycle through them in order, for each new object built.
+* ##### WithSequentialIds()
+  * Sets a numeric property with increasing numbers, from 1 to int.MaxValue.
+* ##### WithEnumerable()
+  * Provide multiple values, and the builder will create the relevant container and put them all onto each new object
+* ##### WithFactory()
+  * Provide a factory method, to set a property to a newly created value each time.
+* ##### WithBuilder()
+  * Like `WithFactory`, but the factory is specifically the `.Build()` method of the provided `Builder`.
+* ##### WithBuilt()
+  * Like `WithBuilder`, but the appropriate `Builder<T>` is found automatically from any classes tagged with `BuilderFactory`, so you don't need to provide any second argument at all ... it just works it all out.
+* ##### _`IEnumerable` variations of `WithFactory()`
+  * The above 3 variations on `WithFactory()` have overloads which simplify the work if the property in question implements `IEnumerable<T>`. Rather than defining a Factory that builds the whole `IEnumerable<T>`, so provide one which simply builds `T`s and the `IEnumerable<>` portion will get worked out for you. The resultant `IEnumerable<T>` will contain 3 elements.
```

Comment:
No; will have another go at the docs.

Reply:
* The above 3 variations on `WithFactory()` have overloads which take effect if the property being set is an `IEnumerable<T>`. In that case, rather than defining a Factory that builds the whole `IEnumerable<T>`, you provide one which simply builds `T`s and the `IEnumerable<>` portion will get worked out for you. The resultant `IEnumerable<T>` will contain 3 elements.